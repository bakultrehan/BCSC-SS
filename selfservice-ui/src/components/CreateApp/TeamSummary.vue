/** * TeamSummary component */

<template>
  <v-card>
    <Loading v-if="isLoading" />
    <v-toolbar dense class="bc-subtitle-2" dark v-else>
      <v-card-title>
        {{ $t('summaryPage.teamTitle') }}
        <v-spacer></v-spacer>
        <v-icon small class="ml-3" @click="$router.push(`/project/${id}/team`)">mdi-pencil</v-icon>
      </v-card-title>
    </v-toolbar>

    <v-list dense class="px-5" v-if="!isLoading">
      <template v-for="(member, index) in team">
        <v-list-item :key="member.id">
          <v-list-item-content class="align-self-start pr-30">
            {{
            $t(`summaryPage.${rolesList[member.role]}`)
            }}
            <span
              class="small-hint pad-50"
              v-html="$t(`summaryPage.${rolesList[member.role]}Hint`)"
            ></span>
          </v-list-item-content>
          <v-list-item-content class="align-end">
            <div>
              <v-icon small class="mr-1">mdi-account</v-icon>
              {{ member.firstName }}
              {{ member.lastName }}
            </div>
            <div v-if="member.phone !== ''" class="ml-6">{{ member.phone }}</div>
            <div v-if="member.email !== ''" class="ml-6">{{ member.email }}</div>
          </v-list-item-content>
        </v-list-item>
        <v-divider :key="`divider-${member.id}`" v-if="index < team.length - 1"></v-divider>
      </template>
    </v-list>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Prop, Watch } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import { projectRolesList } from '@/constants/enums';
import Loading from '@/Atomic/Loading/Loading.vue';

const TeamModule = namespace('TeamRolesModule');

@Component({
  components: { Loading }
})
export default class TeamSummary extends Vue {
  @Prop({ default: 0 })
  public id!: number;

  @TeamModule.Getter('getTeamList')
  public team!: any;
  @TeamModule.Action('loadTeam')
  public loadTeam!: any;

  private isLoading: boolean = true;
  private rolesList: any = projectRolesList;

  @Watch('team')
  private onTeamChanged(val: any) {
    this.isLoading = false;
  }

  private mounted() {
    this.isLoading = true;
    this.loadTeam(this.id);
  }
}
</script>
