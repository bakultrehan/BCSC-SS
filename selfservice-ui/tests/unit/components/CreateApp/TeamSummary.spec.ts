import { shallowMount, mount } from '@vue/test-utils';
import TeamSummary from '@/components/CreateApp/TeamSummary.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
let vuetify: any;
vuetify = new Vuetify();

describe('TeamSummary.vue', () => {
  const store = new Vuex.Store({
    modules: {
      TeamRolesModule: {
        namespaced: true,
        state: {},
        getters: {
          getTeamList: jest.fn(() => {
            return {
              team: [
                {
                  email: 'e@e.com',
                  firstName: 'fname',
                  id: 1,
                  lastName: 'lname',
                  phone: '09870987654321',
                  role: 2
                },
                {
                  email: 'bb@test.com',
                  firstName: 'john',
                  id: 17,
                  lastName: 'bb',
                  phone: '987654321',
                  role: 3
                },
                {
                  email: 'user@yopmail.com',
                  firstName: 'John',
                  id: 1,
                  lastName: 'Doe',
                  phone: '',
                  role: 1
                }
              ]
            };
          })
        },
        actions: {
          loadTeam: jest.fn()
        }
      },
      SharedModule: {
        namespaced: true,
        state: {},
        getters: {
          isRedirectFromSummaryPage: jest.fn()
        },
        actions: {
          redirectFromSummaryPage: jest.fn()
        }
      }
    }
  });

  const mountFunction = (options: any) => {
    return mount(TeamSummary, {
      store,
      vuetify,
      mocks: { $t: jest.fn(() => { }) }, // tslint:disable-line
      ...options
    });
  };

  it('renders props when passed with gettors', () => {
    const teamSummary = mountFunction({});

    expect(teamSummary.element).toMatchSnapshot();
  });

  it('render data', () => {
    const projectSummary = mountFunction({});
    projectSummary.setData({ isLoading: false });
    expect(projectSummary.element).toMatchSnapshot();
  });
});

describe('TeamSummary.vue', () => {
  it('renders props when passed', () => {
    const store = new Vuex.Store({
      modules: {
        TeamRolesModule: {
          namespaced: true,
          state: {},
          getters: {
            getTeamList: jest.fn()
          },
          actions: {
            loadTeam: jest.fn()
          }
        },
        SharedModule: {
          namespaced: true,
          state: {},
          getters: {
            isRedirectFromSummaryPage: jest.fn()
          },
          actions: {
            redirectFromSummaryPage: jest.fn()
          }
        }
      }
    });
    const teamSummary = shallowMount(TeamSummary, {
      vuetify,
      store,
      mocks: { $t: jest.fn(() => { }) } // tslint:disable-line
    });

    expect(teamSummary.element).toMatchSnapshot();
  });
});
