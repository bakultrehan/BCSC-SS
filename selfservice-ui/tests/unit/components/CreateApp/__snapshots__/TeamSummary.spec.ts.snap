// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamSummary.vue render data 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <header
    class="bc-subtitle-2 v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--dense"
    style="height: 48px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 48px;"
    >
      <div
        class="v-card__title"
      >
        
      
      
        <div
          class="spacer"
        />
         
        <i
          class="v-icon notranslate ml-3 v-icon--link mdi mdi-pencil theme--dark"
          role="button"
          style="font-size: 16px;"
          tabindex="0"
        />
      </div>
    </div>
  </header>
   
  <div
    class="v-list px-5 v-sheet v-sheet--tile theme--light v-list--dense"
    role="list"
  >
    <div
      class="v-list-item theme--light"
      role="listitem"
      tabindex="-1"
    >
      <div
        class="v-list-item__content align-self-start pr-30"
      >
        
          
          
        <span
          class="small-hint pad-50"
        />
      </div>
       
      <div
        class="v-list-item__content align-end"
      >
        <div>
          <i
            aria-hidden="true"
            class="v-icon notranslate mr-1 mdi mdi-account theme--light"
            style="font-size: 16px;"
          />
          
            
            
          
        </div>
         
        <div
          class="ml-6"
        >
          
        </div>
         
        <div
          class="ml-6"
        >
          
        </div>
      </div>
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`TeamSummary.vue renders props when passed 1`] = `
<v-card-stub
  loaderheight="4"
  tag="div"
>
  <loading-stub />
   
  <!---->
</v-card-stub>
`;

exports[`TeamSummary.vue renders props when passed with gettors 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="container"
  >
    <div
      class="layout d-flex align-center justify-center row wrap"
    >
      <div
        class="mt-10"
        col="4"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          class="v-progress-linear loading v-progress-linear--rounded theme--light"
          role="progressbar"
          style="height: 6px;"
        >
          <div
            class="v-progress-linear__background primary"
            style="opacity: 0.3; left: 0%; width: 100%;"
          />
          <div
            class="v-progress-linear__buffer"
          />
          <div
            class="v-progress-linear__indeterminate v-progress-linear__indeterminate--active"
          >
            <div
              class="v-progress-linear__indeterminate long primary"
            />
            <div
              class="v-progress-linear__indeterminate short primary"
            />
          </div>
        </div>
         
        <span />
      </div>
    </div>
  </div>
   
  <!---->
</div>
`;
